lib/internal/test_runner/fake_timers.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const {
+  DateNow,
+  SafeMap,
+  Symbol,
+  globalThis,
+} = primordials;
+
+class Timers {
+  constructor() {
+    this.timers = new SafeMap();
+
+    this.setTimeout = this.#createTimer.bind(this, false);
+    this.clearTimeout = this.#clearTimer.bind(this);
+    this.setInterval = this.#createTimer.bind(this, true);
+    this.clearInterval = this.#clearTimer.bind(this);
+  }
+
+  #createTimer(isInterval, callback, delay, ...args) {
+    const timerId = Symbol('kTimerId');
+    const timer = {
+      id: timerId,
+      callback,
+      time: DateNow() + delay,
+      interval: isInterval,
+      args,
+    };
+    this.timers.set(timerId, timer);
+    return timerId;
+  }
+
+  #clearTimer(timerId) {
+    this.timers.delete(timerId);
+  }
+
+}
+
+let realSetTimeout;
+let realClearTimeout;
+let realSetInterval;
+let realClearInterval;
+
+class FakeTimers {
+  constructor() {
+    this.fakeTimers = {};
+    this.isEnabled = false;
+    this.now = DateNow();
+  }
+
+  tick(time = 0) {
+
+    // if (!this.isEnabled) {
+    //   throw new Error('you should enable fakeTimers first by calling the .enable function');
+    // }
+
+    this.now += time;
+    const timers = this.fakeTimers.timers;
+
+    for (const timer of timers.values()) {
```

Comment:
This doesn't seem entirely right? (timers can be called multiple times in a single tick).

Also note Node timers have a bunch of other methods to cancel them like .ref/unref, return objects and can be `.refresh`ed.